tests/functional/object/rgw/test_provider_client.py

```diff
@@ -0,0 +1,76 @@
+import logging
+import pytest
+
+from ocs_ci.framework import config
+from ocs_ci.framework.pytest_customization.marks import (
+    tier1,
+    red_squad,
+    rgw,
+    runs_on_provider,
+    provider_client_ms_platform_required,
+)
+from ocs_ci.ocs import constants
+from ocs_ci.ocs.bucket_utils import sync_object_directory
+
+log = logging.getLogger(__name__)
+
+
+@pytest.fixture
+def return_to_original_context():
+    """
```

Comment:
This fixture looks useful and generic, are you sure you want to use it only for rgw tests?